regression/tests/whitelabel/course_enrollment_test.py

```diff
@@ -133,13 +132,10 @@ def is_course_added_to_dashboard(self):
         t_end = time.time() + TIME_OUT_LIMIT
         # Run the loop for a pre defined time
         while time.time() < t_end:
-            try:
-                if not self.dashboard_page.is_course_present(self.course_id):
-                    raise CourseNotFoundException
+            if self.dashboard_page.is_course_present(self.course_id):
                 return True
-            except CourseNotFoundException:
-                time.sleep(WAIT_TIME)
-                self.dashboard_page.visit()
+            else:
+                continue
```

Comment:
We do need to do that refresh (via visit, or if there's a page object method that will do a browser refresh) during the wait cycle.

I think we should use a promise rather than writing our own loop. 
Something like this pseudo-code:
```
def foo:
    if self.dashboard_page.is_course_present(self.course_id):
        return True
    self.dashboard_page.visit()
    return False

EmptyPromise(foo, "Course is present on dashboard").fulfill()    
```